### PR TITLE
Make it build with ghc-9.10

### PR DESCRIPTION
**Pull request by erikd**
# Description

Make it build with ghc-9.10

# Checklist

### Quality
* [x] Commit sequence makes sense and have useful messages, see [ref][contrib#git-history].
* [ ] New tests are added and existing tests are updated.
* [ ] Self-reviewed the PR.

### Maintenance
* [ ] Linked an [issue][link-issue] or added the PR to the current sprint of [`ouroboros-network`][project] project.
* [ ] Added labels.
* [ ] Updated changelog files.
* [ ] The documentation has been properly updated, see [ref][contrib#documentation].

[project]: https://github.com/orgs/IntersectMBO/projects/5/views/1
[link-issue]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=
[contrib#git-history]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#git-history
[contrib#documentation]: https://github.com/IntersectMBO/ouroboros-network/blob/master/CONTRIBUTING.md#documentation

### CODE REVIEW

File: cabal.project
Comment (erikd) [inline cabal.project:58]: Should probably wait until this is fixed before this can be merged.

Comment (erikd) [reply cabal.project]: That has been removed.

File: quickcheck-monoids/test/Main.hs
Comment (erikd) [inline quickcheck-monoids/test/Main.hs:45]: This has already been disabled for the `Arbitrary Any` instance.

**Review by coot — APPROVED** [review]
LGTM, thanks @erikd.

**Pull request merged**
